packages/@react-aria/focus/src/focusSafely.ts

```diff
@@ -24,12 +24,12 @@ export function focusSafely(element: FocusableElement) {
   // the page before shifting focus. This avoids issues with VoiceOver on iOS
   // causing the page to scroll when moving focus if the element is transitioning
   // from off the screen.
-  const ownerDocument = getOwnerDocument(element);
+  const ownerDocument = getRootNode(element);
   if (getInteractionModality() === 'virtual') {
-    let lastFocusedElement = ownerDocument.activeElement;
+    let lastFocusedElement = ownerDocument?.activeElement;
```

Comment:
why aren't we using getDeepActiveElement here and just below?

Reply:
You are right about it; using `getDeepActiveElement` is better, especially when it is related to nested shadow roots.